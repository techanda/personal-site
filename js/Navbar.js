import React from 'react'
import { Link } from 'react-router'
import { Navbar, Nav, NavItem } from 'react-bootstrap'

const Landing = React.createClass({
  fadeNavBar (selector) {
    var fade = (selector, start, end) => {
      var element = document.querySelector(selector)
      var opacity = start
      var id = setInterval(() => {
        if ((opacity >= end && start < end) || (opacity <= end && start > end)) {
          clearInterval(id)
        } else {
          start < end ? opacity = opacity + 0.1 : opacity = opacity - 0.1
          if (start < end) {
            element.querySelector('.text-fade').classList.remove('fadeout')
          } else {
            element.querySelector('.text-fade').classList.add('fadeout')
          }
          element.style.background = `rgba(16, 16, 16, ${opacity})`
        }
      }, 50)
    }
    var navbarStatus = false
    document.querySelector(selector).style.background = 'rgba(16, 16, 16, 0.0)'

    window.addEventListener('scroll', (event) => {
      if (window.scrollY > 100) {
        if (!navbarStatus) {
          fade(selector, 0.0, 1.0)
          navbarStatus = true
        }
      } else {
        if (navbarStatus) {
          fade(selector, 1.0, 0.0)
          navbarStatus = false
        }
      }
    })
  },
  componentDidMount () {
    //  navbar animation
    this.fadeNavBar('.slide-top')
    // Handle local hash anchors
    document.querySelectorAll('nav li[role=presentation]').forEach((element) => {
      var target = element.querySelector('a').getAttribute('href')
      if (target.startsWith('#')) {
        element.addEventListener('click', () => {
          document.querySelector(`${target}`).scrollIntoView(
            {
              behavior: 'smooth',
              block: 'start'
            }
          )
        })
      }
    })
  },
  render () {
    return (
      <Navbar inverse collapseOnSelect fixedTop className='slide-top' id='top'>
        <Navbar.Header>
          <Navbar.Brand className='text-fade fadeout'>
            <Link to='#top'>
              {/* {'{'}<span className='name'>tim.ramsier</span>{'}'} */}
              <img className='brandImg' src='/public/img/tr_monogram.png' />
            </Link>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav pullRight>
            <NavItem eventKey={1} href='#aboutme'>About Me</NavItem>
            <NavItem eventKey={2} href='#myvalues'>My Values</NavItem>
            <NavItem eventKey={3} href='#'>My Skills</NavItem>
            <NavItem eventKey={4} href='#'>Contact Me</NavItem>
          </Nav>
        </Navbar.Collapse>
      </Navbar>
    )
  }
})

export default Landing
