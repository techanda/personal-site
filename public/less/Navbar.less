@keyframes buttonUnderline {
  from { width: 0%; }
    to { width: 100%;}
}

@keyframes linkColorChange {
  from { color: @base-color-3; }
    to { color: @base-color-2; }
}
.whiteout {
  color: @base-color-2!important;
}

.navbar {
  border: @base-color-4!important;
  border-radius: 0!important;
  font-family: @special-font;
  margin-bottom: 0!important;
  a.navbar-brand {
    color: @base-color-2;
    font-size: 2em;
    .name {
      color: @base-color-1;
      transition: color .5s;
    }
    &:hover {
      .name {
        color: @base-color-0;
      }
    }
  }
  ul.navbar-nav {
    li {
        a {
            color: @base-color-3;
        }
      &:hover {
        a {
          color: @base-color-2;
          animation-name: linkColorChange;
          animation-duration: .5s;
        }
      }
    }

    li::after {
      width:0%;
      height:4px;
      background: @base-color-1;
      content: "";
      display: block;
    }
    li:hover {
      background: lighten(@base-color-4, 10%);
      &::after {
        width:100%;
        animation-name: buttonUnderline;
        animation-duration: .25s;
      }
    }
  }
}
