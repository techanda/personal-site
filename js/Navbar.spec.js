/* globals test expect */

import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Navbar from './Navbar'

test('Navbar snapshot test', () => {
  const component = shallow(<Navbar />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
