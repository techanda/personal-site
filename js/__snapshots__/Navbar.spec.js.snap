exports[`test Navbar snapshot test 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  className="slide-top slide-paused"
  collapseOnSelect={true}
  fixedTop={true}
  inverse={true}>
  <NavbarHeader>
    <NavbarBrand>
      <Link
        activeClassName=""
        activeOnlyWhenExact={false}
        activeStyle={Object {}}
        className=""
        isActive={[Function isActive]}
        replace={false}
        style={Object {}}
        to="#">
        {
        <span
          className="name text-whiteout whiteout">
          tim.ramsier
        </span>
        }
      </Link>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={true}
      stacked={false}>
      <NavItem
        active={false}
        disabled={false}
        eventKey={1}
        href="#aboutme">
        About Me
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        eventKey={2}
        href="#">
        My Skills
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        eventKey={3}
        href="#">
        Contact Me
      </NavItem>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
